src/sql-parser/tests/testdata/lexer

```diff
@@ -174,3 +174,25 @@ Parse error:
 1e10.1
     ^^
 extra token after expression
+
+# Test that unexpected unicode characters do not cause the error message to
+# crash when rendered...
+parse-statement
+SELECT ’hello’
+----
+error:
+Parse error:
+SELECT ’hello’
+       ^^^
+unexpected character in input: ’
+
+# ...though we completely mishandle the width of unicode chars when positioning
+# the caret at the moment.
+parse-statement
+SELECT '’fancy quotes inside a string are ok’’’’' ’
+----
+error:
+Parse error:
+SELECT '’fancy quotes inside a string are ok’’’’' ’
```

Comment:
This parses fine in postgres because it renames the string column to <smart quote>:

```
mjibson=# SELECT '’fancy quotes inside a string are ok’’’’' ’;
                    ’                     
------------------------------------------
 ’fancy quotes inside a string are ok’’’’
(1 row)

```

Reply:
(I'm ok merging this PR even still. Wanted to point this out in case you wanted to fix this.)

Reply:
Ugh! Good catch. Will do in a follow up PR so you can review.